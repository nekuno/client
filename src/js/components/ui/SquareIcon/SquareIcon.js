import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './SquareIcon.scss';

export default class SquareIcon extends Component {

    static propTypes = {
        icon           : PropTypes.string,
        size           : PropTypes.oneOf(['xx-small', 'x-small', 'small', 'medium', 'large', 'answer']).isRequired,
        disabled       : PropTypes.bool,
        background     : PropTypes.string,
        color          : PropTypes.string,
        fontSize       : PropTypes.string,
        onClickHandler : PropTypes.func,
        border         : PropTypes.string
    };

    handleClick() {
        if (this.props.onClickHandler) {
            this.props.onClickHandler();
        }
    }

    render() {
        const {icon, size, disabled, background, color, fontSize, border} = this.props;
        let className = styles.squareIcon + ' ' + styles[size];
        className = disabled ? styles.disabled + ' ' + className : className;

        return (
            <div className={className} style={{background: background, color: color, border: border}} onClick={this.handleClick.bind(this)}>
                <div className={styles.icon + ' icon icon-' + icon} style={{fontSize: fontSize}}/>
            </div>
        );
    }
}