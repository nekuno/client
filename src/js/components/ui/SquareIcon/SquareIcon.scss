@import '../../../../scss/_partials/colors';

.square-icon {
  text-align: center;
  margin: auto;
  cursor: pointer;


  &.disabled {
    cursor: default;
    opacity: 0.5;
  }

  &.xx-small {
    height: 23px;
    width: 23px;

    .icon {
      font-size: 18px;
      line-height: 23px;
      font-weight: bold;
    }
  }
}