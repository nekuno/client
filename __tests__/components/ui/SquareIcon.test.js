import React from 'react';
import SquareIcon from '../../../src/js/components/ui/SquareIcon/SquareIcon.js';
import styles from '../../../src/js/components/ui/SquareIcon/SquareIcon.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test SquareIcon component', () => {
    const mockFn = jest.fn();
    const squareicon = shallow(<SquareIcon onClickHandler={mockFn}>Lorem ipsum</SquareIcon>);

    it('should be defined', () => {
        expect(squareicon).toBeDefined();
    });
    it('should render correctly', () => {
        expect(squareicon).toMatchSnapshot();
    });
    it('SquareIcon click event', () => {
        squareicon.find(`.${styles.squareicon}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
});
