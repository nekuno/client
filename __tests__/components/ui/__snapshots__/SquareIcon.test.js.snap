// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SquareIcon component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SquareIcon
    onClickHandler={[MockFunction]}
  >
    Lorem ipsum
  </SquareIcon>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Lorem ipsum",
      "className": "squareicon",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Lorem ipsum",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem ipsum",
        "className": "squareicon",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Lorem ipsum",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
